compiler/lib/src/main/scala/analysis/Semantics/Event.scala

```diff
@@ -35,14 +45,59 @@ object Event {
         )
         else Right(())
       }
+      def getEveryIntervalValue(every: AstNode[Ast.Expr]) = {
+        val loc = Locations.get(every.id)
+        val Value.AnonStruct(intervalValue) = Analysis.convertValueToType(a.valueMap(every.id), Type.AnonStruct(
+          Map(
+            ("seconds", Type.U32),
+            ("useconds", Type.U32),
+          )
+        ))
+        def getNonZeroMember(member: String, maxValue: Option[Int] = None) = {
+          val Value.PrimitiveInt(v, Type.PrimitiveInt.U32) = Analysis.convertValueToType(
+            intervalValue.get(member).get,
+            Type.U32
+          )
+
+          if (v < 0) {
+            Left(SemanticError.InvalidIntValue(
+              loc, v, s"$member may not be negative"
+            ))
+          } else {
+            maxValue match {
+              case Some(m) => if v >= m then
+                Left(SemanticError.InvalidIntValue(
+                  loc, v, s"$member must be smaller than $m"
+                )) else Right(v.intValue)
+              case None => Right(v.intValue)
+            }
+          }
+        }
+        for {
+          seconds <- getNonZeroMember("seconds")
```

Comment:
The seconds value must fit in a U32.

Reply:
Actually the spec says it has to fit in an I32, and the implementation reflects that. But maybe we should make it a U32, since that is what F Prime uses? If so we need to update the spec and the implementation. In Scala we could use a Long for this, or a BigInt.

Reply:
I just updated the spec.